### PR TITLE
Misc AI fixes

### PR DESCRIPTION
**Pull request by Dilvish-fo**
For details, please note the individual commits.

@Vezzra, while some of these might be most accurately considered AI tweaks and from one point of view not strictly AI bug fixes, I'm inclined to view any weakness in the AI as an AI bug, and I'd like to flag this whole set as cherry-pick for the prospective bug-fix release.

### CODE REVIEW

File: default/python/AI/ProductionAI.py [hunk 1]
```diff
@@ -1410,8 +1410,31 @@ def generate_production_orders():
...
+                queued_troop_ships <= 2):  # todo get max from character module or otherwise calculate
+            best_design_id, best_design, build_choices = best_ships[PriorityType.PRODUCTION_INVASION]
+            loc = random.choice(build_choices)
+            retval = fo.issueEnqueueShipProductionOrder(best_design_id, loc)
+            if retval != 0:
+                per_turn_cost = (float(best_design.productionCost(empire.empireID, loc))/best_design.productionTime(
+                    empire.empireID, loc))
+                avail_pp -= per_turn_cost
+                debug("adding %d extra trooper at location %s to production queue: %s; per turn production cost %.1f\n",
+                      ship_number, PlanetUtilsAI.planet_string(loc), best_design.name, per_turn_cost)
```
Comment (Morlic-fo) [inline default/python/AI/ProductionAI.py]: Seems there is always a single ship enqueued here, not ```ship_number```.

**Review by Morlic-fo — CHANGES_REQUESTED** [review]
Didn't playtest to verify impact of changes but overall looks reasonable. There are however a few issues regarding the changes to ```assess_planet_threat```, see below.

File: default/python/AI/AIDependencies.py [hunk 2]
```diff
@@ -239,6 +239,14 @@
 PANOPTICON_DETECTION_BONUS = 10
 # </editor-fold>
 
+# <editor-fold desc="Native Tech/Defense related specials">
+# (special_name, defense&shield_bonus)
+TECH_NATIVE_SPECIALS = [
+    ("MODERATE_TECH_NATIVES_SPECIAL", 10),
+    ("HIGH_TECH_NATIVES_SPECIAL", 30),
```
[the remark this answers is not in the document]
Comment (Dilvish-fo) [reply default/python/AI/AIDependencies.py]: OK, done

File: default/python/AI/AIstate.py [hunk 3]
```diff
@@ -327,18 +327,34 @@ def __report_system_defenses(self):
...
+        init_shields = planet.initialMeterValue(fo.meterType.shield)
+        next_shields = planet.currentMeterValue(fo.meterType.shield)  # always assumes regen will occur
         max_shields = planet.currentMeterValue(fo.meterType.maxShield)
-        current_defense = planet.currentMeterValue(fo.meterType.defense)
+        init_defense = planet.initialMeterValue(fo.meterType.defense)
+        next_defense = planet.currentMeterValue(fo.meterType.defense)  # always assumes regen will occur
         max_defense = planet.currentMeterValue(fo.meterType.maxDefense)
-        shields = min(max_shields, current_shields + 2 * sighting_age)  # TODO: base off regen tech
-        defense = min(max_defense, current_defense + 2 * sighting_age)  # TODO: base off regen tech
+        shield_regen_turns = 0 if shield_regen_blocked else sighting_age
```
[the remark this answers is not in the document]
Comment (Dilvish-fo) [reply default/python/AI/AIstate.py]: Hmm, yes, that had evolved over a period of time & it does seem that part wound up unneeded; it's cleared out now.

File: default/python/AI/AIstate.py [hunk 4]
```diff
@@ -327,18 +327,34 @@ def __report_system_defenses(self):
...
+        for special, bonus in TECH_NATIVE_SPECIALS:
+            if special in planet.specials and sighting_age > 0:
+                max_shields = max(max_shields, bonus)
+                max_defense = max(max_defense, bonus)
+                next_shields, init_shields = bonus, bonus
+                next_defense, init_defense = bonus, bonus
```
[the remark this answers is not in the document]
Comment (Dilvish-fo) [reply default/python/AI/AIstate.py]: OK, done

File: default/python/AI/AIstate.py [hunk 5]
```diff
@@ -534,11 +557,17 @@ def __update_system_status(self):
                 planet = universe.getPlanet(pid)
                 if not planet:
                     continue
-                prating = self.assess_planet_threat(pid, sighting_age=current_turn - partial_vis_turn)
+                sighting_age = current_turn-get_partial_visibility_turn(pid)
```
[the remark this answers is not in the document]
Comment (Dilvish-fo) [reply default/python/AI/AIstate.py]: OK, done

File: default/python/AI/ProductionAI.py (hunk 1 above)
Comment (Dilvish-fo) [reply default/python/AI/ProductionAI.py]: Hmm, yes, cleaned up the log message

**Review by Morlic-fo — APPROVED** [review]

**Comment by Vezzra** [comment]
@Dilvish-fo 

> while some of these might be most accurately considered AI tweaks and from one point of view not strictly AI bug fixes, I'm inclined to view any weakness in the AI as an AI bug, and I'd like to flag this whole set as cherry-pick for the prospective bug-fix release.

Ok, reasonable.

**Pull request merged**